### ISSUE TITLE
[BUG] "File not found" after correctly consuming it

### ISSUE THREAD

**Issue by Crocmagnon** [paperless-ngx/paperless-ngx#167]
**Describe the bug**
After correctly consuming all 38 files I put in the consume folder, paperless generates a failed task for 22 of them saying it doesn't find them.

I added the files via webdav from the macOS Finder.

All the dropped files were correctly imported, I just have 22 useless failed task generating noise.

**To Reproduce**
Drop an "important" amount of files in the consume folder and watch it unfold. Important in my case being 38, but I also reproduced it with less although I didn't noted the amount.

**Expected behavior**
The files are imported and no failed task is generated after.

**Screenshots**
<img width="312" alt="CleanShot 2022-02-27 at 08 32 57@2x" src="https://user-images.githubusercontent.com/3630554/155873051-005af70d-112e-40b3-8ff1-9a7e13f35ae7.png">

**Webserver logs**
Here is a sample from one failed task:
```
08:21:17 [Q] ERROR Failed [2021-07-23 facture.pdf] - 2021-07-23 facture.pdf: Cannot consume /usr/src/paperless/src/../consume/2021-07-23 facture.pdf: File not found. : Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django_q/cluster.py", line 432, in worker
    res = f(*task["args"], **task["kwargs"])
  File "/usr/src/paperless/src/documents/tasks.py", line 74, in consume_file
    document = Consumer().try_consume_file(
  File "/usr/src/paperless/src/documents/consumer.py", line 195, in try_consume_file
    self.pre_check_file_exists()
  File "/usr/src/paperless/src/documents/consumer.py", line 86, in pre_check_file_exists
    self._fail(
  File "/usr/src/paperless/src/documents/consumer.py", line 70, in _fail
    raise ConsumerError(f"{self.filename}: {log_message or message}")
documents.consumer.ConsumerError: 2021-07-23 facture.pdf: Cannot consume /usr/src/paperless/src/../consume/2021-07-23 facture.pdf: File not found.
```

**Relevant information**
 - Host OS of the machine running paperless: Ubuntu 20.04 for the host, paperless runs in Docker
 - Browser: any
 - Version: 1.5.0 as per the footer, I'm currently running the latest paperless-ngx docker image.
 - Installation method: docker
 - Any configuration changes you made in `docker-compose.yml`, `docker-compose.env` or `paperless.conf`: none relevant.

**Comment by lippoliv** [comment]
@Crocmagnon yes I agree, that inotify can trigger the import multiple times (e.g. file changed on slow network or something like that)

But I think the core problem is concurrency. So another solution could be, to keep track which files paperless already is working on (filename would be enough at this moment) and do not start working on that file a second time, even if inotify is triggering the process.

It can also be checksum, which would also prevent spending computation time on multiple file uploads with the same file?

**Comment by lippoliv** [comment]
@qcasey could this may be fixed with #483?

**Comment by tdiekel** [comment]
[#483](https://github.com/paperless-ngx/paperless-ngx/pull/483) did not fix this issue.

I installed paperless-ngx today for the first time. Tried it with 10 random pdfs and got the error for all. I copied the files via SMB from my desktop into the consume folder. 

#### Log example
```
[2023-02-14 16:53:27,120] [INFO] [paperless.consumer] Consuming Rechnung_R200153825.pdf
...
[2023-02-14 16:53:32,739] [INFO] [paperless.consumer] Document 2021-02-23 Rechnung_R200153825 consumption finished
[2023-02-14 16:53:34,074] [ERROR] [paperless.consumer] Cannot consume /usr/src/paperless/consume/Rechnung_R200153825.pdf: File not found.
```

#### Relevant information
Host OS: Synology (DSM 7.1.1) with Portainer (2.17.0)
Version: 1.12.2
Installation method: [docker-compose.portainer.yml](https://github.com/paperless-ngx/paperless-ngx/blob/main/docker/compose/docker-compose.portainer.yml)
Configuration changes: none
